int32 joint
float32 radians
bool planOnly
bool confirmation
